src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -334,7 +345,7 @@ private static Interop.Sys.OpenFlags PreOpenConfigurationFromOptions(FileMode mo
                 }
 
                 fileLength = status.Size;
-                filePermissions = (Interop.Sys.Permissions)(status.Mode & (int)Interop.Sys.Permissions.Mask);
+                filePermissions = (UnixFileMode)(status.Mode & (int)PermissionMask);
```

Comment:
We have an assert in System.Native that the enum values in Interop.Sys.Permissions match the POSIX values:

[https://github.com/dotnet/runtime/blob/1b4a9c47ba8e4ad53562e003240527b738b0e7d1/src/native/libs/System.Native/pal_io.c#L106-L121](https://github.com/dotnet/runtime/blob/1b4a9c47ba8e4ad53562e003240527b738b0e7d1/src/native/libs/System.Native/pal_io.c#L106-L121)

1. It might be good to add the StickyBit value to this set of asserts, now that we have it in public API.
2. I wonder if we should just delete Interop.Sys.Permissions, and use UnixFileMode everywhere.

Reply:
> I wonder if we should just delete Interop.Sys.Permissions, and use UnixFileMode everywhere.

That makes sense.

We'll also be replacing the `public` `System.Formats.Tar.TarFileMode` by `UnixFileMode`.

I can do it as part of this PR, or defer to a separate PR.